wagtail/admin/views/pages/edit.py

```diff
@@ -332,6 +333,9 @@ def dispatch(self, request, page_id):
 
         self.page = self.real_page_record.get_latest_revision_as_object()
         self.parent = self.page.get_parent()
+        self.scheduled_page = (
+            self.scheduled_revision and self.scheduled_revision.as_object()
+        )
```

Comment:
Shouldn’t this be using the method you introduced so we avoid the repetition?

```suggestion
        self.scheduled_page = self.real_page_record.get_scheduled_revision_as_object()
```